### ISSUE TITLE
ParseUser.currentUser returning null after having logged in

### ISSUE THREAD

**Issue by RyanBuss01** [parse-community/Parse-SDK-Flutter#592]
I am currently trying to set up authentication and when I reset the application "ParseUser.currentUser(); function returns null even after I had logged in.

there are no errors of any kind and logging in works fine but when I restart the app it doesn't find a user 
(also no errors in flutter doctor)

for the rest of my code it's almost identical to the back4app flutter documentation found [here](https://www.back4app.com/docs/flutter/parse-sdk/users/flutter-current-user)

_pubspec.yaml:_

```
dependencies:
  parse_server_sdk: ^2.1.0
```

_main.dart:_

```
void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  final keyApplicationId = 'xxxxxx';
  final keyParseServerUrl = 'https://parseapi.back4app.com';
  final keyClientkey = 'xxxxx';

  await Parse().initialize(
      keyApplicationId,
      keyParseServerUrl,
    clientKey: keyClientkey,
    debug: true
  );
  runApp(MyApp());
}
```

_MyApp:_ 

```
class MyApp extends StatelessWidget {
  Future<bool> hasUserLogged() async {
    print('future called');
    ParseUser currentUser = await ParseUser.currentUser() as ParseUser;
    if (currentUser == null) {
      return false;
    }
    //Validates that the user's session token is valid
    final ParseResponse parseResponse =
    await ParseUser.getCurrentUserFromServer(
        currentUser.get<String>('sessionToken'));

    if (!parseResponse.success) {
      print('call failed');
      //Invalid session. Logout
      await currentUser.logout();
      return false;
    } else {
      print('user found');
      return true;
    }
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter - Parse Server',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: FutureBuilder<bool>(
          future: hasUserLogged(),
          builder: (context, snapshot) {
            switch (snapshot.connectionState) {
              case ConnectionState.none:
              case ConnectionState.waiting:
                return Scaffold(
                  body: Center(
                    child: Container(
                        width: 100,
                        height: 100,
                        child: CircularProgressIndicator()),
                  ),
                );
              default:
                if (snapshot.hasData && snapshot.data) {
                  print('to User');
                  return UserPage();
                } else {
                  print('to Login');
                  return LoginPage();
                }
            }
          }),
    );
  }
}

```

**Comment by RodrigoSMarques** [comment]
```dart
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running Xcode build...
Xcode build done.                                           17,4s
Debug service listening on ws://127.0.0.1:60948/If53Ine-oVg=/ws
Syncing files to device iPhone 12 Pro Max...
flutter: ╭-- Parse Request
flutter: curl -X GET -H 'X-Parse-Session-Token: r:7de007b7981d4418e4d9df60e88fd44c' -H 'user-agent: Flutter Parse SDK 2.1.0' -H 'X-Parse-Application-Id: xxxxxxx' -H 'X-Parse-Client-Key: xxxxxxx' https://parseapi.back4app.com/users/me

 https://parseapi.back4app.com/users/me
flutter: ╰--
flutter: ╭-- Parse Response
Class: _User
Function: ParseApiRQ.currentUser
Status Code: 200
Payload: {"className":"_User","objectId":"Z4HxWMQXX2","createdAt":"2021-03-26T23:30:46.801Z","updatedAt":"2021-03-26T23:39:35.493Z","sessionToken":"r:7de007b7981d4418e4d9df60e88fd44c","username":"rodrigosmarques","email":"rodrigosmarques@example.com","emailVerified":true,"ACL":{"*":{"read":true},"Z4HxWMQXX2":{"read":true,"write":true}}}
╰--

```

**Comment by RyanBuss01** [comment]
I figured out the problem, I changed my dependencies to:

```
dependencies:
  parse_server_sdk_flutter: ^2.1.0
```

and Instead of the original sdk I imported:

```
import 'package:parse_server_sdk_flutter/parse_server_sdk.dart';
```
I'm not sure why this worked but I'm going to close this issue since it's fixed

**Comment by RodrigoSMarques** [comment]
There was a bug in version 2.0.1 that was fixed in version 2.1.0. There is an issue where this was documented.
